src/test/java/test/GoogleMailSaveNewMailAsDraftAndDeleteTest.java

```diff
@@ -32,26 +58,29 @@ public void googleMailSaveMailAsDraftTest() {
         String password = "@Tyrmandyr1!";
         String emailAddressee = "al.hlebka@gmail.com";
         String emailSubject = "Message from Webdriver";
-        String emailBody = "Hello Aliaksei,\nthis message is generated by Webdriver.\nBest regards,\nAliaksei.";
+        JavascriptExecutor javascriptExecutor = (JavascriptExecutor) driver;
 
         GoogleMailHomePage homePage = new GoogleLoginPage(driver).loginToGoogleMail(login, password);
 
-        Assert.assertEquals(driver.getCurrentUrl(), homePage.getHomepageUrl(), "Login failed");
+//        Assert.assertEquals(driver.getCurrentUrl(), homePage.getHomepageUrl(), "Login failed");
+        String currentUrl = (String) javascriptExecutor.executeScript("return document.location.href;");
```

Comment:
Move it out of this layer. It should not be placed in the dough.

Reply:
Method getCurrentPageUrl() created in GoogleMailHomePage class.